### PR TITLE
UX: makes avatar non interactive in thread participants list

### PR DESCRIPTION
**Pull request by jjaffeux**
It was slightly surprising to have a user card show when clicking on a thread item list.

More over this commit does:
- moves `chat/user-avatar` to `chat-user-avatar` and converts it to gjs
- moves `chat/thread/participants` to `chat-thread-participants`
- rewrite the `toggleCheckIfPossible` modifier to only be applied when selecting messages, it prevents the click event to collide with the click of avatars in regular messages

The reason for moving files to top level namespace is that 1) thanks to gjs I will be able to remove `chat-` 2) I noticed that this way of organizing files actually makes things hard for designers as it's not super obvious what the file name is going to be.

### CODE REVIEW

**Review by ZogStriP — APPROVED** [review]

File: plugins/chat/assets/javascripts/discourse/components/chat-user-avatar.gjs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import { renderAvatar } from "discourse/helpers/user-avatar";
+import { htmlSafe } from "@ember/template";
+import concatClass from "discourse/helpers/concat-class";
+
+export default class ChatUserAvatar extends Component {
+  <template>
+    <div
+      class={{concatClass "chat-user-avatar" (if this.isOnline "is-online")}}
+    >
+      {{log this.interactive}}
```
Comment (ZogStriP) [inline plugins/chat/assets/javascripts/discourse/components/chat-user-avatar.gjs:12]: Missed that one?

File: plugins/chat/assets/javascripts/discourse/components/chat-user-avatar.gjs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  @service chat;
+
+  get avatar() {
+    return htmlSafe(
+      renderAvatar(this.args.user, { imageSize: this.avatarSize })
+    );
+  }
+
+  get interactive() {
+    console.log(this.args.interactive);
```
Comment (ZogStriP) [inline plugins/chat/assets/javascripts/discourse/components/chat-user-avatar.gjs:36]: That one too?

File: plugins/chat/spec/fabricators/chat_fabricator.rb [hunk 3]
```diff
@@ -63,7 +63,7 @@
 Fabricator(:chat_message_without_service, class_name: "Chat::Message") do
   user
   chat_channel
-  message { Faker::Lorem.paragraph_by_chars(number: 500).gsub("...", "…") }
+  message { Faker::Lorem.paragraph_by_chars(number: 500).gsub("...", "…").gsub("..", "…") }
```
Comment (ZogStriP) [inline plugins/chat/spec/fabricators/chat_fabricator.rb:66]: Shorter, but not sure if clearer 🤔 

```suggestion
  message { Faker::Lorem.paragraph_by_chars(number: 500).gsub(/[.]{2,3}/, "…") }
```

Comment (jjaffeux) [reply plugins/chat/spec/fabricators/chat_fabricator.rb:66]: yes :/ I think I prefer the other one but thanks

**Pull request merged**